sycl/include/sycl/ext/oneapi/bfloat16.hpp

```diff
@@ -32,6 +32,27 @@ Bfloat16StorageT bfloat16ToBits(const bfloat16 &Value);
 bfloat16 bitsToBfloat16(const Bfloat16StorageT Value);
 } // namespace detail
 
+static detail::Bfloat16StorageT from_float_fallback(const float &a) {
+#if defined(__NVPTX__)
+  if (a != a)
+    return 0xffc1;
+#elif defined(__AMDGCN__)
+  if (sycl::isnan(a))
+    return 0xffc1;
+#else
+  if (std::isnan(a))
+    return 0xffc1;
+#endif
```

Comment:
Before this PR, we would use `if (a != a)` to check for NaN when running on the device and we would use `std::isnan` when running on the host.

Using `sycl::isnan` over `if (a != a)` makes sense because `sycl::isnan` is guaranteed to be available on device code.

Why don't we use `sycl::isnan` also on NVPTX?

The code above does not make it clear that `std::isnan` can only be used on host code.  Can we use `__SYCL_DEVICE_ONLY__` to make this more clear?  Or, can we just use `sycl::isnan` in all cases?  That function is also available on host code.

Reply:
Hi, @gmlueck 
I also thought using sycl::isnan should be OK but not sure whether or not we had any special reason to use "a!=a" for NV targets, if so, let me use sycl::isnan for all targets.
Thanks very much.